testsuite/tests/lib-array/test_array.ml

```diff
@@ -177,3 +177,16 @@ let a : int array =
 [%%expect{|
 val a : int array = [|2; 4; 6; 8|]
 |}]
+
+let a = Array.init_matrix 2 3 (fun i j -> ref (10*i+j));;
+a.(0).(0) := 99;;
+a (* other cells are unchanged *);;
```

Comment:
This test of absence of unexpected sharing makes much more sense for `make_matrix`, which I suppose is not previously tested somewhere in this file? If so, I would suggest writing a test for `make_matrix` that does this modification check, and a usual test for `init_matrix` that just makes it evident that the result is different in all positions.